Samples/src/OracleEmulation/Operations.qs

```diff
@@ -0,0 +1,162 @@
+    // `PermutationOracle.Register` call in the driver for its implementation.
+    operation HalfAnswer(x: Qubit[], y: Qubit[]) : Unit {
+        body intrinsic;
+        adjoint intrinsic;
+    }
+
+    // Define a simple permutation function that is used below to create
+    // another oracle.
+    function DoubleAnswerFunc(x: Int, y: Int) : Int {
+        return 84 ^^^ y;
+    }
+
+    // Measure and print the result.
+    operation MeasureAndDisplay(message: String, register: Qubit[]) : Unit {
+        let answer = MeasureInteger(LittleEndian(register));
+        Message(message + $"{answer}.");
+    }
+
+    // # Summary
+    // Here we demonstrate the use of three simple oracles. Each oracle ignores
+    // the content of the first register and XOR's a constant number into the
+    // second register.
+    //
+    // # Input
+    // ## oracle
+    // A quantum operation that implements an oracle
+    //      $O: \ket{x}\ket{y} \rightarrow \ket{x}\ket{f(x, y)}$.
+    operation RunConstantOracles (oracle: ((Qubit[], Qubit[]) => Unit)) : Unit {
+        Message("Querying the oracles...");
+
+        // Prepare a one-qubit register `x` and an eight-qubit register `y`.
+        // Since all the oracles here ignore x, its length and state do not
+        // matter.
+        using (qubits = Qubit[9]) {
+            let x = qubits[0];
+            let y = qubits[1..8];
+            H(x);
+
+            // Apply an oracle that was passed explicitly by the C# driver.
+            oracle([x], y);
+            MeasureAndDisplay("The answer is ", y);
+
+            // Apply an oracle that was declared as `intrinsic` above and
+            // implemented in the C# driver.
+            HalfAnswer([x], y);
+            MeasureAndDisplay("Half the answer is ", y);
+
+            // Apply an oracle defined in terms of a Q# permutation function.
+            EmulateOracle(DoubleAnswerFunc, [x], y);
+            MeasureAndDisplay("Twice the answer is ", y);
+
+            // Apply an oracle to a superposition in y.
+            for(i in 1..5) {
+                H(y[7]);
+                // Before the oracle is queried, the state of the y register is
+                //      $\ket{y} = \ket{0} + \ket{128}$.
+                // The oracle will map this to
+                //      $\ket{y'} = \ket{42 \oplus 0} + \ket{42 \oplus 128} = \ket{42} + \ket{170}$.
+                oracle([x], y);
+                MeasureAndDisplay("The answer might be ", y);
+            }
+
+            Reset(x);
+        }
+    }
+
+
+    //////////////////////////////////////////////////////////////////////////
+    // Emulated arithmetic operations ////////////////////////////////////////
+    //////////////////////////////////////////////////////////////////////////
+
+    // Prepare two `LittleEndian` registers in a computational basis state.
+    operation PrepareSummands(qubits: Qubit[], numbers: Int[]) : (LittleEndian, LittleEndian) {
```

Comment:
In general, the [Q# style guide](https://docs.microsoft.com/quantum/contributing/style-guide) suggests that qubit arguments should follow classical arguments. I'd also suggest that if the `qubits` input is always interpreted as two summands, that `(Qubit[], Qubit[])` may be a better type to represent that.

Reply:
Agree, done!